### PR TITLE
Blacklisted packages are now added to the pkgs_kept_back list - fix for #52 ?

### PR DESCRIPTION
**Pull request by jayaich**
This pull request simply adds the relevant blacklisted package to the pkgs_kept_back list. To do this though it also required the calling functions to be passed the list.
Tested and seems to work okay when only blacklisted packages are available for updating, and for when there are blacklisted and 'ordinary' packages available (packages which are neither black nor white listed).
(One trivial typo also included.)

### CODE REVIEW

File: unattended-upgrade [hunk 1]
```diff
@@ -395,7 +395,7 @@ def match_whitelist_string(whitelist, origin):
             res &= fnmatch.fnmatch(origin.codename, value)
         else:
             raise UnknownMatcherError(
-                "Unknown whitelist entry for macher '%s' (token '%s')" % (
+                "Unknown whitelist entry for matcher '%s' (token '%s')" % (
```
Comment (mvo5) [inline unattended-upgrade]: Nice catch, thank you!

**Comment by mvo5** [comment]
A test for this would be great (or just updating an existing one). Please let me know if that would be something you would be interested in doing :)

**Comment by jayaich** [comment]
I'm just taking a look at the tests. I see that whitelisting is checked, but perhaps not blacklisting. I'm no python guru, so I'll see if I can do anything but it may well come back your way.

**Comment by jayaich** [comment]
As the main part of this fix (#52) just ensures that blacklisted packages are in the pkgs_held_back list, and this is used within the summary email message, I have modified the 'test/test_mail.py' file to include a couple of tests. Both tests ensure that the subject line is correct, and that the relevant text line is in the mail message. The second test ensures that it is still all correct even if there are no packages updated at that time.

**Comment by mvo5** [comment]
@jayaich Yay, thanks a lot! \o/

**Pull request merged**